### PR TITLE
Fixed cache backends bug #11322

### PR DESCRIPTION
**Pull request by ifsnow**
It's a PR to save number 0 in cache. (issue #11322)

### CODE REVIEW

**Comment by sergeyklay** [comment]
Can you please provide a small test?

**Comment by ifsnow** [comment]
@sergeyklay

I think it's just a issue related to checking if variable is null.

```
if !content {
    // if content is null
} else {
    // if content is not null
}
```

In this case, if content parameter is 0 or "0", It works as if content is null.
It's not the operation that user wanted. (user wanted to save "0")

apc.zep and memory.zep already use `content === null` condition to check the parameter.

**Comment by sergeyklay** [comment]
In any case, you must show that your PR has solved the issue. Now it is impossible to check without accepting your PR. A little test would have shown -  issue is solved or not without accepting the PR to the main codebase

**Comment by ifsnow** [comment]
@sergeyklay
I added some codes in `testDataFileCache` function of `CacheTest.php`.
It would failed at this test in 2.0.x branch.

**Comment by sergeyklay** [comment]
Can you please rebase onto `2.0.x`

**Comment by ifsnow** [comment]
@sergeyklay I have rebased onto 2.0.x

**Comment by sergeyklay** [comment]
Fine.
How about:
```php
$cache->save('test-zero-data', 0); // int, not string as described in issue #11322

// or

$cache->save('test-zero-data', false);
```

can you please also provide these test cases

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,8 +9,7 @@
 - Fixed `Phalcon\Translate\Adapter\Gettext::exists` bug[#11310](https://github.com/phalcon/cphalcon/issues/11310) related to the wrong returned value (always true)
 - Fixed `Phalcon\Translate\Adapter\Gettext::setLocale` bug[#11311](https://github.com/phalcon/cphalcon/issues/11311) related to the incorrect setting locale
 - Added ability to persistent connection in `Phalcon\Queue\Beanstalk::connect`
-- Fixed `Phalcon\Http\Response::redirect` bug[#11324](https://github.com/phalcon/cphalcon/issues/11324). Incorrect initialization local array of status codes
-
+- Fixed `Phalcon\Http\Response::redirect` bug[#11324](https://github.com/phalcon/cphalcon/issues/11324). Incorrect initialization local array of status codes- Fixed cache backends bug[#11322](https://github.com/phalcon/cphalcon/issues/11322) related to saving number 0
```
Comment (sergeyklay) [inline CHANGELOG.md]: Put your changes at new line

**Comment by ifsnow** [comment]
@sergeyklay added another test cases & changed wrong `CHANGELOG.md`

**Comment by sergeyklay** [comment]
Thanks

**Pull request merged**